supporting_scripts/generate_code_cov_table/generate_code_cov_table.py

```diff
@@ -227,30 +256,41 @@ def main(argv):
     args = parser.parse_args(argv)
     if args.verbose:
         logging.getLogger().setLevel(logging.DEBUG)
-    if args.password is None:
-        args.password = getpass.getpass("Please enter your Bamboo password: ")
+    if args.token is None:
+        args.token = getpass.getpass("Please enter your Github token: ")
     if args.branch_name is None:
         args.branch_name = get_branch_name()
         logging.info(f"Using current branch: {args.branch_name}")
     if args.build_id is None:
-        args.build_id = get_latest_build_id(args.username, args.password, args.branch_name)
-        logging.info(f"Using latest build ID: {args.build_id}")
+        logging.info("Using latest build ID")
 
     file_changes = get_changed_files(args.branch_name, args.base_branch_name)
     client_file_changes, server_file_changes = filter_file_changes(file_changes)
 
+    headers = {
+        'Authorization': f'token {args.token}',
+        'Accept': 'application/vnd.github.v3+json'
+    }
+
+    artifacts = get_artifacts_of_the_last_completed_run(headers, args.branch_name, args.build_id)
+
+    client_coverage_zip_bytes = download_and_extract_zip(get_coverage_artifact_for_key(artifacts, client_tests_key),
+                                                         headers)
+    server_coverage_zip_bytes = download_and_extract_zip(get_coverage_artifact_for_key(artifacts, server_tests_key),
+                                                         headers)
```

Comment:
is there a reason why this is formatted like this? otherwise I would reformat it:

```suggestion
    client_coverage_zip_bytes = download_and_extract_zip(get_coverage_artifact_for_key(artifacts, client_tests_key),  headers)
    server_coverage_zip_bytes = download_and_extract_zip(get_coverage_artifact_for_key(artifacts, server_tests_key), headers)
```